src/main/scala/org/hammerlab/guacamole/pileup/Pileup.scala

```diff
@@ -150,8 +150,7 @@ object Pileup {
    * @return A [[Pileup]] at the given locus.
    */
   def apply(reads: Seq[MappedRead], referenceName: String, locus: Long, referenceContigSequence: ContigSequence): Pileup = {
-    //TODO: Is this call to overlaps locus necessary?
-    val elements = reads.filter(_.overlapsLocus(locus)).map(PileupElement(_, locus, referenceContigSequence))
+    val elements = reads.map(PileupElement(_, locus, referenceContigSequence))
```

Comment:
Good point about docstring, done!

> Were you seeing overhead in all the `overlaps` checks?

Nope, it's just an extra O(n) over all reads at a spot that may be one of the "tighter loops" in the codebase, so I was scrutinizing it for that reason.

Having thought about it more, though:
- all calling code makes sure to only throw reads at `Pileup.apply` that it knows should be there, and with good reason, since in the cases we currently handle, you wouldn't want to build a `Pileup` out of a bunch of reads whose proximity to the locus in question is not known.
- the `PileupElement` ctor calls `advanceToLocus` which throws an assertion error if a read doesn't overlap; I prefer that behavior here since in general we shouldn't be passing irrelevant reads to `Pileup`.